notebooks/config.ipynb

```diff
@@ -964,12 +971,831 @@
     "<!--cell 54-->"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "249ae7a9-1ad3-4401-98b0-bc3c433f22f4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on function compare in module uwtools.api.config:\n",
+      "\n",
+      "compare(config_1_path: Union[pathlib.Path, str], config_2_path: Union[pathlib.Path, str], config_1_format: Optional[str] = None, config_2_format: Optional[str] = None) -> bool\n",
+      "    Compare two config files.\n",
+      "\n",
+      "    Recognized file extensions are: ini, nml, sh, yaml\n",
+      "\n",
+      "    :param config_1_path: Path to 1st config file.\n",
+      "    :param config_2_path: Path to 2nd config file.\n",
+      "    :param config_1_format: Format of 1st config file (optional if file's extension is recognized).\n",
+      "    :param config_2_format: Format of 2nd config file (optional if file's extension is recognized).\n",
+      "    :return: ``False`` if config files had differences, otherwise ``True``.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.compare)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "908b17fe-ae2d-4964-a659-0481d063c037",
+   "metadata": {},
+   "source": [
+    "Consider the following config files, which have similar values, with the exception of `sent`'s value.\n",
+    "<!--cell 56-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "0218e5de-2c25-4d7a-a7b6-0f05ad81afb2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "&memo\n",
+      "  sender_id = \"{{ id }}\"\n",
+      "  message = \"{{ greeting }}, {{ recipient }}!\"\n",
+      "  sent = .FALSE.\n",
+      "/\n",
+      "----------------------------------------------\n",
+      "&memo\n",
+      "  sender_id = \"{{ id }}\"\n",
+      "  message = \"{{ greeting }}, {{ recipient }}!\"\n",
+      "  sent = .TRUE.\n",
+      "/\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/base-config.nml\n",
+    "echo ----------------------------------------------\n",
+    "cat fixtures/config/alt-config.nml "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4361a7c-9c16-4bc0-b3e6-985d85f1450c",
+   "metadata": {},
+   "source": [
+    "`compare()` returns `True` if the configs contain identical key-value pairs, and `False` otherwise. If a logger has been initialized, a log is displayed which reports which files are being compared and the values that differ, if any. Files are passed to `config_1_path` and `config_2_path` as a string filename or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Corresponding formats are passed to `config_1_format` and `config_2_format`, with these parameters being optional if the format is recognized.\n",
```

Comment:
```suggestion
    "`compare()` returns `True` if the configs contain identical key-value pairs, and `False` otherwise. If a logger has been initialized, information is logged on which files are being compared and the values that differ, if any. Files are passed to `config_1_path` and `config_2_path` as a string filename or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Corresponding optional formats may be passed using `config_1_format` and `config_2_format` and are only needed if the format suffix is not recognized.\n",
```